internal/runbits/dependencies/changesummary.go

```diff
@@ -1,43 +1,86 @@
 package dependencies
 
 import (
+	"encoding/json"
 	"fmt"
 	"sort"
 	"strconv"
 	"strings"
 
+	"github.com/go-openapi/strfmt"
+
+	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/locale"
 	"github.com/ActiveState/cli/internal/logging"
-	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
 	"github.com/ActiveState/cli/internal/sliceutils"
 	"github.com/ActiveState/cli/pkg/buildplan"
+	"github.com/ActiveState/cli/pkg/platform/model/buildplanner"
 )
 
+type primeable interface {
+	primer.Outputer
+	primer.Auther
+	primer.Projecter
+}
+
 // showUpdatedPackages specifies whether or not to include updated dependencies in the direct
 // dependencies list, and whether or not to include updated dependencies when calculating indirect
 // dependency numbers.
 const showUpdatedPackages = true
 
-// OutputChangeSummary looks over the given build plans, and computes and lists the additional
-// dependencies being installed for the requested packages, if any.
-func OutputChangeSummary(out output.Outputer, newBuildPlan *buildplan.BuildPlan, oldBuildPlan *buildplan.BuildPlan) {
-	requested := newBuildPlan.RequestedArtifacts().ToIDMap()
+func OutputChangeSummary(prime primeable, newCommit *buildplanner.Commit, oldCommit *buildplanner.Commit) error {
+	// Fetch the impact report.
+	beforeExpr, err := json.Marshal(oldCommit.BuildScript())
+	if err != nil {
+		return errs.Wrap(err, "Unable to marshal old buildexpression")
+	}
+
+	afterExpr, err := json.Marshal(newCommit.BuildScript())
+	if err != nil {
+		return errs.Wrap(err, "Unable to marshal buildexpression")
+	}
+	bpm := buildplanner.NewBuildPlannerModel(prime.Auth())
+	params := &buildplanner.ImpactReportParams{
+		Owner:      prime.Project().Owner(),
+		Project:    prime.Project().Name(),
+		BeforeExpr: beforeExpr,
+		AfterExpr:  afterExpr,
+	}
+	report, err := bpm.ImpactReport(params)
+	if err != nil {
+		return errs.Wrap(err, "Failed to fetch impact report")
+	}
+
+	buildPlan := newCommit.BuildPlan()
 
+	// Process the impact report, looking for package additions or updates.
+	alreadyInstalledVersions := map[strfmt.UUID]string{}
 	addedString := []string{}
 	addedLocale := []string{}
 	dependencies := buildplan.Ingredients{}
 	directDependencies := buildplan.Ingredients{}
-	changeset := newBuildPlan.DiffArtifacts(oldBuildPlan, false)
-	for _, a := range changeset.Added {
-		if _, exists := requested[a.ArtifactID]; exists {
-			v := fmt.Sprintf("%s@%s", a.Name(), a.Version())
+	for _, i := range report.Ingredients {
+		if i.Before != nil {
+			alreadyInstalledVersions[strfmt.UUID(i.Before.IngredientID)] = i.Before.Version
+		}
+
+		if i.After == nil || !i.After.IsRequirement {
+			continue
+		}
+
+		if i.Before == nil {
```

Comment:
May as well pair this up with the identical conditional on line 64.

Reply:
I would, but they're not identical :scream: